### DISCUSSION TITLE
Support Terminal Commands for WOL (and other broadcasts) to the App

### DISCUSSION THREAD

**Discussion by Nitrousoxide** [seriousm4x/UpSnap#391] [Feature requests]
UpSnap works great as a web interface for sending commands to devices on your network. However there doesn't seem to be a way to use it from the terminal as a WoL relay.

```
/app # ./upsnap admin --help
Manages admin accounts

Usage:
  upsnap admin [command]

Available Commands:
  create      Creates a new admin account
  delete      Deletes an existing admin account
  update      Changes the password of a single admin account

Flags:
  -h, --help   help for admin

Global Flags:
      --dev                    enable dev mode, aka. printing logs and sql statements to the console
      --dir string             the PocketBase data directory (default "/app/pb_data")
      --encryptionEnv string   the env variable whose value of 32 characters will be used 
                               as encryption key for the app settings (default none)

Use "upsnap admin [command] --help" for more information about a command.
/app # 
```

Say, for instance, you are on another subnet from the device you wan to wake.  You must log into the web interface and manually click the button to send a WoL packet.  It does not appear as though you can set up an alias like `fedora-remote-wake="ssh server 'docker exec upsnap upsnapwol device-1'" so you can, with one command, send a wol packet through UpSnap as a proxy.  I am aware there are a few concerns with this.  First, obviously, you don't want just anyone to be able to pass commands to UpSnap to perform actions on your devices like sleep/wake them.  You may need to authenticate the user.  You would also need to provide a list of devices saved to UpSnap and the actions it can perform to them so the user can write the command out.

There are obviously other WoL proxies one could use, but this doesn't seem like it would bloat this project much as it's just providing command line access to the already existing actions its taking when commanded via the web interface.

Totally understand though if you say this is outside the scope of this project.

**Comment by seriousm4x** [comment]
Hi, upsnap is meant to be run as a server application only, that's why there are now cli commands to wake individual devices. That would be one reason to have a separate cli client for upsnap...

However, can't you make use of the [api](https://github.com/seriousm4x/UpSnap/wiki/Rest-API) to wake devices instead of passing commands to the container? Sounds simpler to me.

**Comment by Nitrousoxide** [comment]
Thanks, that's a fine solution I can use instead!  I'll close this issue.
